File: src/org/genyris/test/io/StringFormatStreamTest.java
// Copyright 2008 Peter William Birch <birchb@genyis.org>
//
// This software may be used and distributed according to the terms
// of the Genyris License, in the file "LICENSE", incorporated herein by reference.
//
package org.genyris.test.io;

import junit.framework.TestCase;

import org.genyris.io.ConvertEofInStream;
import org.genyris.io.InStream;
import org.genyris.io.LexException;
import org.genyris.io.StringFormatStream;
import org.genyris.io.StringInStream;
import org.genyris.io.UngettableInStream;

public class StringFormatStreamTest extends TestCase {

    protected void setUp() throws Exception {
        super.setUp();
    }

    private void excerciseSFS(String toparse, String expected) throws LexException {
        InStream ind = new ConvertEofInStream(new StringFormatStream(new UngettableInStream(
                new StringInStream(toparse))));
        StringBuffer result = new StringBuffer();
        while (ind.hasData()) {
            result.append(ind.readNext());
        }
        assertEquals(expected, result.toString());
    }



    public void testSFStreamEmpty() throws LexException {
        excerciseSFS("", "\"\"");
    }
    public void testSFStream2() throws LexException {
        excerciseSFS("1", "\"1\"");
    }
    public void testSFStream3() throws LexException {
        excerciseSFS("1", "\"1\"");
    }
    public void testSFStream4() throws LexException {
        excerciseSFS("12343567890", "\"12343567890\"");
    }
    public void testSFStream5() throws LexException {
        excerciseSFS("#{}", "\"\"\"\"");
    }
    public void testSFStream6() throws LexException {
        excerciseSFS("#{alpha}", "\"\"alpha\"\"");
    }
    public void testSFStream7() throws LexException {
        excerciseSFS("one#{alpha}two", "\"one\"alpha\"two\"");
    }

    public void testSFStream8() throws LexException {
        excerciseSFS("on\"e#{alpha}two", "\"on\\\"e\"alpha\"two\"");
    }

    public void testSFStreamReal() throws LexException {
        excerciseSFS("<img src=\"#{image-url-var}\">", "\"<img src=\\\"\"image-url-var\"\\\">\"");
    }
    public void testSFStreamMultiple() throws LexException {
        excerciseSFS("12#{3}435#{6}78#{9}0", "\"12\"3\"435\"6\"78\"9\"0\"");
    }

}

File: src/org/genyris/test/io/StringFormatStreamTest.kt
// Copyright 2008 Peter William Birch <birchb@genyis.org>
//
// This software may be used and distributed according to the terms
// of the Genyris License, in the file "LICENSE", incorporated herein by reference.
//
package org.genyris.test.io

import junit.framework.Assert
import junit.framework.TestCase
import org.genyris.io.*

class StringFormatStreamTest : TestCase() {
    @kotlin.Throws(Exception::class)
    override fun setUp() {
        super.setUp()
    }

    @kotlin.Throws(LexException::class)
    private fun excerciseSFS(toparse: String, expected: String?) {
        val ind: InStream = ConvertEofInStream(
            StringFormatStream(
                UngettableInStream(
                    StringInStream(toparse)
                )
            )
        )
        val result = StringBuffer()
        while (ind.hasData()) {
            result.append(ind.readNext())
        }
        Assert.assertEquals(expected, result.toString())
    }


    @kotlin.Throws(LexException::class)
    fun testSFStreamEmpty() {
        excerciseSFS("", "\"\"")
    }

    @kotlin.Throws(LexException::class)
    fun testSFStream2() {
        excerciseSFS("1", "\"1\"")
    }

    @kotlin.Throws(LexException::class)
    fun testSFStream3() {
        excerciseSFS("1", "\"1\"")
    }

    @kotlin.Throws(LexException::class)
    fun testSFStream4() {
        excerciseSFS("12343567890", "\"12343567890\"")
    }

    @kotlin.Throws(LexException::class)
    fun testSFStream5() {
        excerciseSFS("#{}", "\"\"\"\"")
    }

    @kotlin.Throws(LexException::class)
    fun testSFStream6() {
        excerciseSFS("#{alpha}", "\"\"alpha\"\"")
    }

    @kotlin.Throws(LexException::class)
    fun testSFStream7() {
        excerciseSFS("one#{alpha}two", "\"one\"alpha\"two\"")
    }

    @kotlin.Throws(LexException::class)
    fun testSFStream8() {
        excerciseSFS("on\"e#{alpha}two", "\"on\\\"e\"alpha\"two\"")
    }

    @kotlin.Throws(LexException::class)
    fun testSFStreamReal() {
        excerciseSFS("<img src=\"#{image-url-var}\">", "\"<img src=\\\"\"image-url-var\"\\\">\"")
    }

    @kotlin.Throws(LexException::class)
    fun testSFStreamMultiple() {
        excerciseSFS("12#{3}435#{6}78#{9}0", "\"12\"3\"435\"6\"78\"9\"0\"")
    }
}
